ASC Release Notes 4.2.0 (#7565)
magento/devdocs: src/extensions/amazon-sales/release-notes/index.md

title: Amazon Sales Channel Release Notes
---
{:.bs-callout .bs-callout-warning}

> Amazon Sales Channel versions 4.0.0 and 4.1.0 are only supported with Magento 2.3.x versions.<br/>Amazon Sales Channel version 4.2.0 is compatible with Magento 2.3.x versions but is only supported with Magento 2.4.x versions.

Reviewer: ```suggestion
Amazon Sales Channel versions 4.0.0 and 4.1.0 are only supported for Magento 2.3.x versions.<br/>Amazon Sales Channel version 4.2.0 is compatible with Magento 2.3.x versions but is only supported for Magento 2.4.x versions.
```

Author: updated

See [Upcoming Releases](https://devdocs.magento.com/release/) for versioning, support, and compatibility.

### v4.2.0

> Amazon Sales Channel 4.2.0 is compatible with versions 2.3.x but only supported in versions 2.4.x of {{site.data.var.ce}}, {{site.data.var.ee}}, and {{site.data.var.ece}}. If you have a previous Amazon Sales Channel version installed and attempt to update your Magento to version 2.4.0, you will be prompted to update the extension before you can complete the Magento update.

Reviewer: ```suggestion
Amazon Sales Channel 4.2.0 is compatible with versions 2.3.x but only supported 2.4.x versions of {{site.data.var.ce}}, {{site.data.var.ee}}, and {{site.data.var.ece}}. If you have a previous Amazon Sales Channel version installed and attempt to update your Magento to version 2.4.0, you will be prompted to update the extension before you can complete the Magento update.
```

Author: changed "in" to "for", removal makes it awkward

This version of Amazon Sales Channel includes a new feature along with improvements and fixes.

> -  {:.new}Amazon Sales Channel has been enhanced to accept text-based address data and match it to standardized address formats, including city, state, and zip code. This enables order and shipping data to synchronize with Amazon without address errors. Example: Shopper inputs the city, state, zip code as `Escondido, californiA 92025-1501`. Amazon Sales Channel imports and matches the data to the standard format as `Escondido, CA 92025`, and then syncs it back to Amazon in this standardized format.

Reviewer: I might consider putting the example on a separate line. Or say something like, "For example, the shopper inputs..." instead of having an incomplete sentence.

Author: valid. changing format to For example,.....

> -  {:.new}Support for PHP 7.4 has been added.

Reviewer: ```suggestion
-  {:.new}Support for PHP 7.4 was added.
```

Reviewer: Try to resist using "be" verbs: has been, will be, etc.

Author: changed format to match other bullets. "Added support for..."

-  {:.fix}<!--CHAN-4334-->The extension has been updated for support with Magento 2.4.x. Previous versions may be compatible with Magento 2.4.x, but are not supported. See [Upcoming releases](https://devdocs.magento.com/release/) for version compatibility. Amazon Sales Channel must be updated to 4.2.0 before the Magento 2.4.0 update can be completed.

> -  {:.fix}<!--CHAN-4431-->Corrected an error causing an "Access Denied" error for UK customers.

Reviewer: Two errors? Is there another way to word this?

Author: corrected: issue caused an error

> -  {:.fix}<!--CHAN-4394-->Corrected an issue preventing the Amazon shipping status from syncing to the corresponding Magento order, thus “locking” the order’s shipping status as `Pending` in Magento and `Unshipped` in Amazon. With the implementation of the new standardized address feature, these shipping status errors have been resolved.

Reviewer: ```suggestion
-  {:.fix}<!--CHAN-4394-->Corrected an issue preventing the Amazon shipping status from syncing to the corresponding Magento order, which locked the order’s shipping status as `Pending` in Magento and `Unshipped` in Amazon. With the implementation of the new standardized address feature these shipping status errors have been resolved.
```

> -  {:.fix}<!--ticket#-->Order synchronization has been updated to ignore failed order imports, thus reducing multiple sync attempts and allowing subsequent imports to process, with order sync requests submitted every 5 minutes. Synchronization errors are still recorded in the error log. Synchronization errors are logged, but marked as "processed" to allow further logging functions. As part of this fix, Amazon Sales Channel now automatically removes excess data collected for orders that fail to create in Magento.

Reviewer: ```suggestion
-  {:.fix}<!--ticket#-->Order synchronization has been updated to ignore failed order imports, reducing multiple sync attempts and allowing subsequent imports to process. Order sync requests are submitted every five minutes. Synchronization errors are still recorded in the error log and are logged, but marked as "processed" to allow further logging functions. As part of this fix, Amazon Sales Channel now automatically removes the excess data collected for failed orders.
```

> -  {:.fix}Error logging has been updated to collect more information for “Uncaught Exception” and extension update errors.

Reviewer: ```suggestion
-  {:.fix}Error logging was updated to collect more information for uncaught exception and extension update errors.
```

Author: corrected.

> -  {:.fix}<!--ticket#-->Corrected an issue causing the initial sync of "Lowest Price" data to fail due to missing "Price" value.

Reviewer: I'm not sure why so many things are in quotation marks, but that's bad practice.

Reviewer: If they represent something in code, or a status, or similar, it should be in code formatting.

Author: corrected with ital

> -  {:.fix}<!--ticket#-->Corrected an issue causing the initial sync of "Lowest Price" data to fail due to missing "Price" value.

Reviewer: ```suggestion
-  {:.fix}<!--ticket#-->Corrected an issue that caused the initial sync of the Lowest Price data to fail due to a missing Price value.
```

Author: changed all "causing" to "that caused".  same for "preventing".

> -  {:.fix}<!--CHAN-4410-->Corrected issues causing errors when the date range field is left blank when filtering your orders view.

Reviewer: ```suggestion
-  {:.fix}<!--CHAN-4410-->Corrected issues which caused errors in blank date range fields when filtering the Orders view.
```

Reviewer: Is Orders view the name of an actual view int he Admin? If so, it should be capped as suggested. If not, just orders view.

Author: the errors were not in the date range field.   the errors occurred because the date range field is left blank.  it is the Amazon orders screen.   reworded as: Corrected issues that caused filtering errors in the _Amazon orders_ view when the date range field is left blank.

> -  {:.fix}<!--CHAN-4439-->Corrected an issue causing stock and fulfillment sync errors related to quantity. The extension has been updated to round down quantity values entered as a decimal before syncing with Amazon. Example: Merchant manually inputs `2.5` for quantity, the extension updates the value to `2` and then syncs the updated quantity with Amazon.

Reviewer: ```suggestion
-  {:.fix}<!--CHAN-4439-->Corrected an issue that caused quantity-related stock and fulfillment sync errors. The extension has been updated to round down quantity values entered as a decimal before syncing with Amazon. For example, when he merchant manually inputs a quantity of `2.5`, the extension updates the value to `2` and then syncs the updated quantity with Amazon.
```

Author: updated.
